src/main/java/com/iota/iri/conf/BaseIotaConfig.java

```diff
@@ -669,6 +755,13 @@ protected void setBelowMaxDepthTransactionLimit(int maxAnalyzedTransactions) {
                 "KPWCHICGJZXKE9GSUDXZYUAPLHAKAHYHDXNPHENTERYMMBQOPSQIDENXKLKCEYCPVTZQLEEJVYJZV9BWU";
 
         //Snapshot
+        boolean LOCAL_SNAPSHOTS_ENABLED = true;
+        boolean LOCAL_SNAPSHOTS_PRUNING_ENABLED = false;
+        int LOCAL_SNAPSHOTS_PRUNING_DELAY = 1000;
+        int LOCAL_SNAPSHOTS_INTERVAL_SYNCED = 10;
+        int LOCAL_SNAPSHOTS_INTERVAL_UNSYNCED = 5000;
+        String LOCAL_SNAPSHOTS_BASE_PATH = "mainnet";
+        int LOCAL_SNAPSHOTS_DEPTH = 500;
```

Comment:
Do we have any sort of "story" backing up these default values?

Reply:
**LOCAL_SNAPSHOTS_INTERVAL_SYNCED = 10**
Taking local snapshots regularly makes the processing faster since it doesnt have to examine so many new transactions so we want to do it relatively often but also not too often since it still consumes as few resources - 5 minutes have proven to be a good value while testing to keep the processing times short (below a few seconds) and still don't bother the node too many times with creating the snapshots.

**LOCAL_SNAPSHOTS_INTERVAL_UNSYNCED = 5000**
When we are syncing we usually process a lot of milestones very rapidly. To not bother the syncing process too much but still have a fast resume point when IRI crashes for example, we create a snapshot every 5000 milestones. In addition to reducing the amount of local snapshots taken and the corresponding load on the node, it also creates a situation where we have a bigger chunk of the old tangle data available most of the time because the transactions only get pruned when a snapshot was taken. This allows us to revert the state of the ledger further back in case we detect a milestone transaction that was skipped previously and continue the syncing without an unrecoverable error - repairing milestones will be included in a future PR.

**int LOCAL_SNAPSHOTS_DEPTH = 500;**
I reduced the snapshot depth back to 500 because it works 99.9% of the time (i only once faced a situation where data was referenced by the coo that was older than 500 milestones. Since we also have a pruning delay now (new feature) which keeps old transactions prior to the taken snapshot., this situation should not be a problem anymore. At the same time a lower snapshot depth allows it to sync new nodes extremely fast.

Reply:
> Since we also have a pruning delay now (new feature) which keeps old transactions prior to the taken snapshot.
So effectively you will be able to reference data up to `LOCAL_SNAPSHOTS_DEPTH + LOCAL_SNAPSHOTS_PRUNING_DELAY`, assuming pruning is enabled.

Reply:
yep